src/lib/sidenav/sidenav.ts

```diff
@@ -316,6 +317,35 @@ export class MdSidenavContainer implements AfterContentInit {
   /** The sidenav child with the `end` alignment. */
   get end() { return this._end; }
 
+  /**
+   * Width of the container at which it breaks
+   * e.g., breakpointWidth="500"
+   */
+  @Input() breakpointWidth: number;
+
+  /** Mode of the breakpoint; either true or false */
+  @Input() breakpointChangeMode: true | false = true;
+  
+  /**
+   * Responsively toggle the sidenav using the breakpoint.
+   * Note that this only works when the window is rezied.
+   * If you resize it some other way, call checkBreakpoint().
+   */
+  @HostListener('window:resize', ['$event'])
```

Comment:
This should be debounced. See https://github.com/angular/material2/blob/master/src/lib/tabs/tab-header.ts#L182-L195